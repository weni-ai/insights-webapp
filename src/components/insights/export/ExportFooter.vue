<template>
  <section class="export-footer">
    <section class="export-footer__format">
      <UnnnicLabel :label="formatLabel" />
      <UnnnicRadio
        :data-testid="'radio-format-xlsx'"
        :modelValue="selectedFormat"
        value=".xlsx"
        @update:model-value="handleFormatChange('.xlsx')"
      >
        {{ '.XLSX' }}
      </UnnnicRadio>
      <UnnnicRadio
        :data-testid="'radio-format-csv'"
        :modelValue="selectedFormat"
        value=".csv"
        @update:model-value="handleFormatChange('.csv')"
      >
        {{ '.CSV' }}
      </UnnnicRadio>
    </section>

    <section class="export-footer__terms">
      <p class="export-footer__terms-warning">
        <UnnnicIcon
          icon="alert-circle-1"
          filled
          next
          scheme="feedback-yellow"
          size="ant"
        />{{ warningTermsText }}
      </p>
      <UnnnicCheckbox
        :modelValue="acceptTerms"
        :textRight="acceptTermsText"
        @update:model-value="handleAcceptTermsChange"
      />
    </section>
  </section>
</template>

<script setup lang="ts">
import {
  UnnnicCheckbox,
  UnnnicIcon,
  UnnnicLabel,
  UnnnicRadio,
} from '@weni/unnnic-system';

type FormatType = '.csv' | '.xlsx';

interface Props {
  selectedFormat: FormatType;
  acceptTerms: boolean;
  formatLabel: string;
  warningTermsText: string;
  acceptTermsText: string;
}

withDefaults(defineProps<Props>(), {
  selectedFormat: '.xlsx',
  acceptTerms: false,
  formatLabel: '',
  warningTermsText: '',
  acceptTermsText: '',
});

const emit = defineEmits<{
  'format-change': [format: FormatType];
  'accept-terms-change': [value: boolean];
}>();

const handleFormatChange = (format: FormatType): void => {
  emit('format-change', format);
};

const handleAcceptTermsChange = (value: boolean): void => {
  emit('accept-terms-change', value);
};
</script>

<style scoped lang="scss">
.export-footer {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__format {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-sm;
  }

  &__terms {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-sm;
  }

  &__terms-warning {
    display: flex;
    gap: $unnnic-spacing-xs;

    color: $unnnic-color-neutral-dark;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }
}
</style>
