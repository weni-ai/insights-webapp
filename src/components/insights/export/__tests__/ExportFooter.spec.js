import { beforeEach, describe, expect, it } from 'vitest';
import { config, mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';

import ExportFooter from '../ExportFooter.vue';

const i18n = createI18n({
  legacy: false,
  locale: 'en',
  messages: { en: {} },
});

config.global.plugins = [i18n];

describe('ExportFooter', () => {
  let wrapper;

  const defaultProps = {
    selectedFormat: '.xlsx',
    acceptTerms: false,
    formatLabel: 'Format',
    warningTermsText: 'Warning message',
    acceptTermsText: 'I accept terms',
  };

  const createWrapper = (props = {}) => {
    return mount(ExportFooter, {
      props: { ...defaultProps, ...props },
      global: {
        stubs: {
          UnnnicRadio: true,
          UnnnicCheckbox: true,
          UnnnicLabel: true,
          UnnnicIcon: true,
        },
      },
    });
  };

  const section = () => wrapper.find('.export-footer');

  beforeEach(() => {
    wrapper = createWrapper();
  });

  describe('Component rendering', () => {
    it('should render main section', () => {
      expect(section().exists()).toBe(true);
    });

    it('should render format section with radio buttons', () => {
      const formatSection = wrapper.find('.export-footer__format');
      expect(formatSection.exists()).toBe(true);
    });

    it('should render terms section', () => {
      const termsSection = wrapper.find('.export-footer__terms');
      expect(termsSection.exists()).toBe(true);
    });
  });

  describe('Props', () => {
    it('should accept selectedFormat prop', () => {
      wrapper = createWrapper({ selectedFormat: '.csv' });
      expect(wrapper.props('selectedFormat')).toBe('.csv');
    });

    it('should accept acceptTerms prop', () => {
      wrapper = createWrapper({ acceptTerms: true });
      expect(wrapper.props('acceptTerms')).toBe(true);
    });

    it('should use default values', () => {
      wrapper = createWrapper({
        formatLabel: undefined,
        warningTermsText: undefined,
        acceptTermsText: undefined,
      });
      expect(wrapper.props('formatLabel')).toBe('');
      expect(wrapper.props('warningTermsText')).toBe('');
      expect(wrapper.props('acceptTermsText')).toBe('');
    });
  });

  describe('Events', () => {
    it('should emit format-change event when format changes', async () => {
      await wrapper.vm.handleFormatChange('.csv');

      expect(wrapper.emitted('format-change')).toBeTruthy();
      expect(wrapper.emitted('format-change')[0]).toEqual(['.csv']);
    });

    it('should emit accept-terms-change event', async () => {
      await wrapper.vm.handleAcceptTermsChange(true);

      expect(wrapper.emitted('accept-terms-change')).toBeTruthy();
      expect(wrapper.emitted('accept-terms-change')[0]).toEqual([true]);
    });

    it('should handle multiple format changes', async () => {
      await wrapper.vm.handleFormatChange('.csv');
      await wrapper.vm.handleFormatChange('.xlsx');

      expect(wrapper.emitted('format-change')).toHaveLength(2);
      expect(wrapper.emitted('format-change')[1]).toEqual(['.xlsx']);
    });
  });

  describe('Component structure', () => {
    it('should have correct CSS classes', () => {
      expect(section().classes()).toContain('export-footer');
    });

    it('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
