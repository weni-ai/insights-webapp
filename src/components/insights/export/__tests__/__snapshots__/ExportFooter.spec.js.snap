// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExportFooter > Component structure > should match snapshot 1`] = `
<section
  class="export-footer"
  data-testid="export-footer"
  data-v-4f08a4d6=""
>
  <section
    class="export-footer__format"
    data-testid="export-footer-format"
    data-v-4f08a4d6=""
  >
    <unnnic-label-stub
      data-v-4f08a4d6=""
      label="Format"
    />
    <div
      class="unnnic-radio-container unnnic-radio-container--md"
      data-testid="radio-format-xlsx"
      data-v-4f08a4d6=""
      data-v-abd1b195=""
    >
      <unnnic-icon-stub
        class="unnnic-radio"
        clickable="false"
        data-v-abd1b195=""
        filled="false"
        icon="radio-selected"
        next="false"
        scheme="brand-weni"
        size="md"
      />
      <span
        class="label"
        data-v-abd1b195=""
      >
        
        .XLSX
        
      </span>
    </div>
    <div
      class="unnnic-radio-container unnnic-radio-container--md"
      data-testid="radio-format-csv"
      data-v-4f08a4d6=""
      data-v-abd1b195=""
    >
      <unnnic-icon-stub
        class="unnnic-radio"
        clickable="false"
        data-v-abd1b195=""
        filled="false"
        icon="radio-default"
        next="false"
        scheme="neutral-cleanest"
        size="md"
      />
      <span
        class="label"
        data-v-abd1b195=""
      >
        
        .CSV
        
      </span>
    </div>
  </section>
  <section
    class="export-footer__terms"
    data-testid="export-footer-terms"
    data-v-4f08a4d6=""
  >
    <p
      class="export-footer__terms-warning"
      data-testid="export-footer-warning"
      data-v-4f08a4d6=""
    >
      <unnnic-icon-stub
        clickable="false"
        data-v-4f08a4d6=""
        filled="true"
        icon="alert-circle-1"
        next="true"
        scheme="feedback-yellow"
        size="ant"
      />
      Warning message
    </p>
    <div
      class="unnnic-checkbox-wrapper"
      data-testid="export-footer-accept-terms"
      data-v-4f08a4d6=""
      data-v-f5f8c62c=""
    >
      <unnnic-icon-stub
        class="unnnic-checkbox"
        clickable="true"
        data-v-f5f8c62c=""
        filled="false"
        icon="checkbox-default"
        next="false"
        scheme="neutral-clean"
        size="md"
      />
      <div
        class="unnnic-checkbox__label unnnic-checkbox__label__right unnnic-checkbox__label__md"
        data-testid="checkbox-text-right"
        data-v-f5f8c62c=""
      >
        I accept terms
      </div>
    </div>
  </section>
</section>
`;
